src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -66,7 +69,11 @@ public void Apply(BsonClassMap classMap)
                     continue;
                 }
 
-                classMap.MapConstructor(ctor);
+                if (ctor.IsPublic && !typeInfo.IsAbstract)
+                {
+                    // we need to save constructorInfo only for public constructors in non abstract classes
+                    classMap.MapConstructor(ctor);
+                }
 
                 anyConstructorsWereMapped = true;
```

Comment:
This variable name is misleading. It can be true even when no constructors were in fact mapped.

Perhaps rename to `anyConstructorsWereFound`?

Reply:
And fix up any other names or comments to standardize on found instead of "mapped"

Reply:
done